### DISCUSSION TITLE
Laser won't burn / M3 command is inverted

### DISCUSSION THREAD

**Discussion by Thibs38** [arkypita/LaserGRBL#1605] [Q&A]
Hi, I recently bought a laser kit from Aliexpress , I followed all of the instructions to set up everything but I'm stuck and I'm going crazy. Basically, I can't get the laser to burn with LaserGRBL.

**The setup:**
The laser is plugged into a small board with a 12V power supply. I then plugged the TTL and Ground pin from the board to the motherboard, which has its own power supply and is connected to the motors / fan / pc with USB. 

![IMG_20210804_235516_resized_20210805_122137875](https://user-images.githubusercontent.com/55209726/128264012-fd9fbfbb-1088-4bcd-aeb5-f2c54017521c.jpg)

![IMG_20210805_000456_resized_20210805_122135691](https://user-images.githubusercontent.com/55209726/128264198-05260a81-3992-49af-a6e9-14826bc92e15.jpg)

The little board that powers the laser has a test button that fully powers the laser, so when idling the laser is not burning:

![IMG_20210804_235454_resized_20210805_122138762](https://user-images.githubusercontent.com/55209726/128263775-b508502f-41df-4466-9a4c-e63801647f40.jpg)

And when I press that button it immediately starts to burn (it is a 20W laser).

![IMG_20210804_235446_resized_20210805_122139260](https://user-images.githubusercontent.com/55209726/128263687-605a4c68-6faa-42bb-a822-55db15073064.jpg)

**The software:**
LaserGRBL is recognizing the laser correctly, and I can move it with no problems. However, I wasn't able to power it more than the idle state. When trying to "print" an image, the laser moves correctly but the power doesn't change. The configuration is set up correctly with laser mode to 1.

![IMG_20210804_235639_resized_20210805_122136155](https://user-images.githubusercontent.com/55209726/128264276-45191a98-5a4f-4481-9cdd-ca4ea3dbd604.jpg)

I even tried a custom button, and surprisingly telling the laser to go max power makes it minimal and vice versa! A bit as if the M3 command was inverted

![IMG_20210804_235609_resized_20210805_122136859](https://user-images.githubusercontent.com/55209726/128264060-6dc9692b-b526-4b9a-8499-80ca749c5018.jpg)

![IMG_20210804_235504_resized_20210805_122138324](https://user-images.githubusercontent.com/55209726/128263954-fdc8e643-ee10-4a23-8367-836795c71db5.jpg)

I tried different versions of LaserGRBL (4.4.1 and 3.8.5), but nothing works. Can anyone help me please?

**Comment by StuartB4** [comment]
Are you sure the laser pwm (yellow) and ground (black) are connected the correct way round on the laser driver module (one with the fan). Yellow is PWM and should go to the S and ground is black and should go to the -
